File: ts-sdk/src/utils/get.forta.chain.id.ts
// returns the scanner-injected chain id to specify which chain should be scanned by this bot
export type GetFortaChainId = () => number | undefined;

export function provideGetFortaChainId(
  fortaChainId: number | undefined
): GetFortaChainId {
  return function getFortaChainId() {
    return fortaChainId;
  };
}

File: py-sdk/src/forta_bot_sdk/utils/get_forta_chain_id.py
from typing import Callable, Optional


# returns the chain id that is injected by the scan node into this bot
GetFortaChainId = Callable[[], Optional[int]]


def provide_get_forta_chain_id(forta_chain_id: Optional[int]):

    def get_forta_chain_id():
        return forta_chain_id

    return get_forta_chain_id
